pandas/tests/io/test_excel.py

```diff
@@ -2557,3 +2557,135 @@ def test_excelwriter_fspath(self):
         with tm.ensure_clean('foo.xlsx') as path:
             writer = ExcelWriter(path)
             assert os.fspath(writer) == str(path)
+
+
+@td.skip_if_no('odf')
+class TestODFReader(SharedItems):
+    def test_get_sheet(self):
+        from pandas.io.excel._odfreader import _ODFReader
+
+        pth = os.path.join(self.dirpath, 'datatypes.ods')
+        book = _ODFReader(pth)
+
+        with pytest.raises(ValueError):
+            book.get_sheet(3.14)
+
+        with pytest.raises(ValueError):
+            book.get_sheet_by_name("Invalid Sheet 77")
+
+        with pytest.raises(IndexError):
+            book.get_sheet_by_index(-33)
+
+        assert len(book.sheet_names) == 1
+        assert book.sheet_names == ['Sheet1']
+
+    def test_read_types(self):
+        """Make sure we read ODF data types correctly
```

Comment:
OK to remove docstring - I don't think we have these for any other tests

Reply:
Some of the docstrings were just restating the function name, but some of them are descriptions of what the test is about. I'd like those more detailed comments to stay, is it ok if they're docstrings or should I turn them into # comments?

Like:
```
    def test_read_invalid_types(self):
        """Make sure we throw an exception when encountering a new value-type

        I had to manually create an invalid ods file by directly
        editing the extracted xml. So it probably won't open in
        LibreOffice correctly.
        """
```

Reply:
Comments are fine though as minimal as possible is preferred